src/contracts/webauthn_account_contract/src/utils.nr

```diff
@@ -19,6 +19,37 @@ fn convert_to_array_to_field(a: [u8; 32]) -> Field {
     res
 }
 
+pub fn prefix_sha256_hash<N>(prefix_length: u8, input: [u8; N]) -> [u8; 32] {
```

Comment:
It feels off, like it should be more like:
```
    if prefix_length == 2 {
        result = std::hash::sha256(prefix(input, [0; 2])) ;
    }
```

Reply:
Repeating `std::hash::sha256` feels less DRY IMO but let's try how it looks. Changed.

Trying to use the latter version results in:
`Message:  Length of array is unknown when evaluating numeric generic`

Reply:
Looks great!

Reply:
Perhaps could be:
```
    if prefix_length == 1 {
        result = prefix(input, [0; 1]);
    }
    if prefix_length == 2 {
        result = prefix(input, [0; 2]);
    }
    if prefix_length == 3 {
        result = prefix(input, [0; 3]);
    }
    if prefix_length == 114 {
        result = prefix(input, [0; 114]);
    }
    if prefix_length == 134 {
        result = prefix(input, [0; 134]);
    }
    if prefix_length == 243 {
        result = prefix(input, [0; 243]);
    }
    std::hash::sha256(result)
```

Reply:
Unfortunately this won't work. `prefix(input, [0; 1])` has a different type than `prefix(input, [0; 2])`

Reply:
I wonder if it could be generic type... Anyhow looks good!